IHP/IDE/CodeGen/Controller.hs

```diff
@@ -27,8 +27,16 @@ instance Controller CodeGenController where
 
     action NewControllerAction = do
         let controllerName = paramOrDefault "" "name"
-        plan <- ControllerGenerator.buildPlan controllerName
-        render NewControllerView { .. }
+        controllerAlreadyExists <- case Text.splitOn "." controllerName of
+            [applicationName, controllerName'] -> doesFileExist $ cs applicationName <> "/Controller/" <> cs controllerName' <> ".hs"
+            [controllerName'] -> doesFileExist $ "Web/Controller/" <> cs controllerName' <> ".hs"
+        if controllerAlreadyExists
```

Comment:
We could write this less nested by using `when`:

```haskell
when controllerAlreadyExists do
    setErrorMessage "..."
    redirect ...
```

Reply:
Done :)